src/background_updates/get_background_updates/v1.rs

```diff
@@ -0,0 +1,95 @@
+//! [GET /_synapse/admin/v1/background_updates/status](https://github.com/element-hq/synapse/blob/master/docs/usage/administration/admin_api/background_updates.md#status)
+
+use std::collections::HashMap;
+use ruma::api::{metadata, request, response, Metadata};
+use serde::{Deserialize, Serialize};
+
+const METADATA: Metadata = metadata! {
+    method: GET,
+    rate_limited: false,
+    authentication: AccessToken,
+    history: {
+        unstable => "/_synapse/admin/v1/background_updates/status",
+    }
+};
+
+#[request]
+#[derive(Default)]
+pub struct Request {}
+
+#[response]
+#[derive(Serialize, Deserialize, PartialEq)]
+pub struct Response {
+    /// Whether the background updates are enabled or disabled.
+    pub enabled: bool,
+
+    /// The current update based on database name.
+    pub current_updates: HashMap<String, CurrentUpdate>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+pub struct CurrentUpdate {
+    /// Name of the update
+    pub name: String,
+
+    /// Total number of processed "items"
+    pub total_item_count: i64,
```

Comment:
Note: one of the background updates is syncing the user directory, which in huge servers could reach to billions. At least, I've seen it reaching a billion. Feel free to correct me here.

Reply:
If this was part of the spec, I would be a bit concerned about this (especially if the value range wasn't called out in the spec). However, it's just the Synapse Admin API which is not as rigorously specified as the official `/_matrix/` endpoints so going by what makes sense realistically is perfectly fine.

Reply:
I set it to i32. I haven't found any specs to how high this can get.

Reply:
Hm, why? The previous type made more sense to me with the context you provided 😅

Or actually, why allow negative values? I think the two most sensible types here would be `u64` and `js_int::UInt` (which would be used if this was in ruma itself).

Reply:
Darn it. Yes, I should have used unsigned here.
I thought it's good to allocate the least memory possible since there was no documentation. You are right, my context should have been my documentation for it. 👍